### ISSUE TITLE
accounts: Accessibility issues

### ISSUE THREAD

**Issue by garrett** [cockpit-project/cockpit#15195]
- [ ] #accounts-list has children in its `<ul>` that are not `<li>` elements. Either this should change from a list item or we should remove the role="presentation" attribute from the `<li>` elements, which cause the element to be disregarded by the accessibility tree.
- [ ] The set password dialog's progress bar (used for password strength) has an invalid ARIA attribute called "aria-valuenow". Also, it is set to an empty string.

**Comment by KKoukiou** [comment]
I can't reproduce the first point. For me it looks like:

``` html
<ul id="accounts-list">
    <li class="cockpit-account" role="presentation">
    </li>
</ul>
```

**Comment by garrett** [comment]
`role="presentation"` is in the <li>, meaning that the element doesn't exist in the accessibility tree. So the a11y tree, which is passed to screen readers, looks like an empty list, like:

```html
<ul id="accounts-list">

</ul>
```

And from the DOM side, it's a non-li element, so it looks like this:

```html
<ul id="accounts-list">
   <invalid_element></invalid_element>
</ul>
```

Neither trees actually have `ul > li`, due to the presentation role. (Although the DOM inspector still shows it, despite the attribute.)

So, in this case, we probably want to remove that `role="presentation"`, or possibly move it elsewhere.

---

Thanks for opening the PF issue!

**Comment by garrett** [comment]
Oh, that PF issue doesn't cover the issue I mentioned here. There's an invalid attribute (even if it is set to nothing). That attribute should not be there, as it does not exist. "aria-valuenow" is not an actual thing. This is a different bug than what you reported upstream.

**Comment by marusak** [comment]
>  "aria-valuenow" is not an actual thing

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-valuenow_attribute

But indeed when the password is empty we should set it to 0 as per "If the current value is not known, the author should not set the aria-valuenow attribute"

**Comment by marusak** [comment]
Being fixed in #15470

**Comment by garrett** [comment]
> https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-valuenow_attribute

Oh! :exploding_head: 

> But indeed when the password is empty we should set it to 0 as per "If the current value is not known, the author should not set the aria-valuenow attribute"

OK!
